File: frontend/lib/models/cow.dart
class Cow {
  final int id;
  final String cowId;
  final String name;
  final String breed;
  final int lactationMonth;
  final String? imagePath;
  final DateTime createdAt;

  Cow({
    required this.id,
    required this.cowId,
    required this.name,
    required this.breed,
    required this.lactationMonth,
    this.imagePath,
    required this.createdAt,
  });

  factory Cow.fromMap(Map<String, Object?> map) {
    return Cow(
      id: map['id'] as int,
      cowId: map['cow_id'] as String,
      name: map['name'] as String,
      breed: map['breed'] as String,
      lactationMonth: (map['lactation_month'] as int),
      imagePath: map['image_path'] as String?,
      createdAt: DateTime.parse(map['created_at'] as String),
    );
  }
}

File: cattle_identify/app/Models/Cow.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Cow extends Model
{
    protected $fillable = [
        'user_id',
        'cow_id',
        'name',
        'breed',
        'lactation_month',
        'weight',
        'previous_disease',
        'image_path',
        'birthdate',
        'embedding',
    ];

    protected $hidden = [
        'embedding',
    ];

    protected $casts = [
        'previous_disease' => 'array',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function feeds()
    {
        return $this->hasMany(CowFeed::class);
    }

    public function identifications()
    {
        return $this->hasMany(CowIdentification::class, 'matched_cow_id');
    }

    public function diseaseDetections()
    {
        return $this->hasMany(DiseaseDetection::class);
    }

    public function birthPredictions()
    {
        return $this->hasMany(AnimalBirthPrediction::class);
    }

    public function behaviorDetections()
    {
        return $this->hasMany(BehaviorDetection::class);
    }

    public function nutritionRecommendations()
    {
        return $this->hasMany(NutritionRecommendation::class);
    }
}